README.md

```diff
@@ -1,2 +1,178 @@
+- `AccountService` with `@Public` service annotation, which makes all methods available for all.
+- `login` method without an annotation. And as a result, it uses `AccountService` scope - `@Public`.
+- `getPublicUserProfile` method with `@RequireRole(Role.Teacher)`. Only teachers are allowed to call this endpoint.
+
+### Auth Header
+
+To get access scope of some roles, you need to have an access token tied to an account with that role.  
+Your access token must be sent in Auth header with a request in the following schema:
+
+```
+Authorization: JWT ${accessToken}
+```
+
+#### Get testing access token
+The easiest way to get an access token for some test purposes is through classtime website:
+1. Go to a [Classtime website](https://www.classtime.com/auth/login) and authorize using any method (e.g. with Google).
+2. Open "classtime.com" cookies in the developer console of the browser. (see e.g. [how to do it in Google Chrome](https://developer.chrome.com/docs/devtools/application/cookies/#open))
+3. Copy a value of a cookie with the name: "service-jwt-0". It is your's teacher account access token.
+
+## JSON requests
+
+JSON requests are the easiest point to start. However, we are recommending to use binary requests.  
+Note: **POST is the only supported method**
+
+**Public AccountService.login call**:
+```
+curl -v https://www.classtime.com/service/public/Account/login \
+    -X POST \
+    -H 'Content-Type: application/json' \
+    -H 'Accept: application/json,*/*' \
+    -d '{ provider: "Google", role: "Teacher" }'
+```
+```
+Response: { "redirectUrl": "https://accounts.google.com..." }
+```
+
+**Restricted AccountService.getPublicUserProfile call**:
+Pre-requirement: you must know your account id and access token. You could obtain it following [this small guide](https://github.com/gopollock/public-api/blob/main/flow-examples/my-account-id.md).
+
+```
+curl -v https://www.classtime.com/service/public/Account/getPublicUserProfile \
+    -X POST \
+    -H 'Content-Type: application/json' \
+    -H 'Accept: application/json,*/*' \
+    -H 'Authorization: JWT ${yourTeacherAccessToken}' \
+    -d '{ account_id: ${yourAccountId} }'
+```
+```
+Response: {
+    "email": "you@email.com",
+    userProfile: {
+      "firstName": "John",
+      "lastName": "White",
+      "picture": "",
+    }
+    ...,
+}
+```
+
+## Binary requests
+
+Pre-requirement for binary requests are generated protobuf. They would help you to serialize requests and deserialize responses.  
+The full process of protobuf generation and usage is described in [generating protos guide](https://github.com/gopollock/public-api/blob/main/generate-protos.md).
+
+**Public AccountService.login call**:
+```
+curl -v https://www.classtime.com/service/Account/login \
+    -X POST \
+    -H 'Content-Type: application/protobuf' \
+    -H 'Accept: application/protobuf,*/*' \
+    -d '' \
+    -o output.bin | tee output.bin
```

Comment:
That seems wrong, -o already writes to a file no?

Reply:
Yes, it is enough to have `-o` or `tee`.
Both write to file. And `tee` additionally logs written content in terminal